src/hipscat/inspection/visualize_catalog.py

```diff
@@ -29,118 +47,247 @@ def _read_point_map(catalog_base_dir):
-    """Get the healpy plotting method for a specified projection string
+    orders = np.array([p.order for p in pixels])
+    ipix = np.array([p.pixel for p in pixels])
+    order_map = orders.copy()
+    fig, ax = plot_healpix_map(
+        order_map, projection=projection, title=plot_title, ipix=ipix, depth=orders, cbar=False, **kwargs
+    )
+    col = ax.collections[0]
+    plt.colorbar(
+        col,
+        boundaries=np.arange(np.min(order_map) - 0.5, np.max(order_map) + 0.6, 1),
+        ticks=np.arange(np.min(order_map), np.max(order_map) + 1),
+        label="order",
+    )
+    return fig, ax
+
+
+def cull_from_pixel_map(depth_ipix_d: Dict[int, Tuple[np.ndarray, np.ndarray]], wcs, max_split_depth=7):
+    """Modified from mocpy.moc.plot.culling_backfacing_cells.from_moc
+
+    Create a new MOC that do not contain the HEALPix cells that are backfacing the projection."""
+    depths = list(depth_ipix_d.keys())
+    min_depth = min(depths)
+    max_depth = max(depths)
+    ipixels, vals = depth_ipix_d[min_depth]
+
+    # Split the cells located at the border of the projection
+    # until at least the max_split_depth
+    max_split_depth = max(max_split_depth, max_depth)
+
+    ipix_d = {}
+    for depth in range(min_depth, max_split_depth + 1):
+        if depth < 3:
+            too_large_ipix = ipixels
+            too_large_vals = vals
+        else:
+            ipix_lon, ipix_lat = cdshealpix.vertices(ipixels, depth)
+
+            ipix_lon = ipix_lon[:, [2, 3, 0, 1]]
+            ipix_lat = ipix_lat[:, [2, 3, 0, 1]]
+            ipix_vertices = SkyCoord(ipix_lon, ipix_lat, frame=ICRS())
+
+            # Projection on the given WCS
+            xp, yp = skycoord_to_pixel(coords=ipix_vertices, wcs=wcs)
+            _, _, frontface_id = backface_culling(xp, yp)
+
+            # Get the pixels which are backfacing the projection
+            backfacing_ipix = ipixels[~frontface_id]  # backfacing
+            backfacing_vals = vals[~frontface_id]
+            frontface_ipix = ipixels[frontface_id]
+            frontface_vals = vals[frontface_id]
+
+            ipix_d.update({depth: (frontface_ipix, frontface_vals)})
+
+            too_large_ipix = backfacing_ipix
+            too_large_vals = backfacing_vals
+
+        next_depth = depth + 1
+        ipixels = np.array([], dtype=ipixels.dtype)
+        vals = np.array([], dtype=vals.dtype)
+
+        if next_depth in depth_ipix_d:
+            ipixels, vals = depth_ipix_d[next_depth]
+
+        # split too large ipix into next order, with each child getting the same map value as parent
+
+        too_large_child_ipix = np.repeat(too_large_ipix << 2, 4) + np.tile(
+            np.array([0, 1, 2, 3]), len(too_large_ipix)
+        )
+        too_large_child_vals = np.repeat(too_large_vals, 4)
+
+        ipixels = np.concatenate((ipixels, too_large_child_ipix))
+        vals = np.concatenate((vals, too_large_child_vals))
+
+    return ipix_d
+
+
+def plot_healpix_map(
```

Comment:
HIgh-level question. If I select some FOV, the color bar still shows the same range as for the whole plot. I'd probably prefer to have better color resolution when I zoom. Maybe it is just a documentation issue about `norm`/`cmap` arguments.

See an example with `plot_points` and DES DR2 catalog:
<img width="843" alt="image" src="https://github.com/user-attachments/assets/e40f5cc1-de4a-44d6-908c-fb64f9af5449">
<img width="801" alt="image" src="https://github.com/user-attachments/assets/e884e604-ccad-47a0-8c72-c8c80c3ea068">

Reply:
Yeah, we should cull the pixels outside of the fov, but I was lazy and I didn't think about this use case so I didn't think it would impact anything other than performance. But this is a good idea to add.

Reply:
Ok this has been updated! Pixels outside the FOV now get culled